### PR TITLE
chore: improve k8s installation

### PR DESCRIPTION
**Pull request by zll600**
- Use StatefulSet instead of Deployment.
- Add ingress configuration.

### CODE REVIEW

**Comment by rebelice** [comment]
Need @h3n4l to review

File: content/docs/get-started/self-host.md [hunk 1]
```diff
@@ -304,6 +272,42 @@ helm -n <YOUR_NAMESPACE> \
...
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    nginx.ingress.kubernetes.io/ssl-redirect: 'true'
+    # https://kubernetes.github.io/ingress-nginx/user-guide/miscellaneous/#websockets
+    nginx.ingress.kubernetes.io/proxy-read-timeout: '3600'
+    nginx.ingress.kubernetes.io/proxy-send-timeout: '3600'
+  name: demo-ingress
```
Comment (h3n4l) [inline content/docs/get-started/self-host.md:288]: bytebase-ingress?

**Review by h3n4l — APPROVED** [review]
LGTM

File: content/docs/get-started/self-host.md (hunk 1 above)
Comment (zll600) [reply content/docs/get-started/self-host.md]: ok

**Pull request merged**